### ISSUE TITLE
How to achieve the high performance?

### ISSUE THREAD

**Issue by ivy94419** [ruotianluo/ImageCaptioning.pytorch#59]
I have trained the **_fc_model_** for 30 epochs, and the hyper-parameters are set the same as the default.
But when testing, I only achieve Bleu-4 = 0.25 in my 1000 validation images in COCO, but I used your provided pre-trained model and achieved Bleu-4=0.324

Bleu_4 0.25786956526235716
METEOR 0.221778680264552
CIDEr 0.785944241965
ROUGE_L 0.494435436025
Bleu_2 0.5063064959709507
Bleu_1 0.6833226219296441
Bleu_3 0.3629853091429978

How to reproduce the high performance as you achieved?

**Comment by ruotianluo** [comment]
Use the full 5000 val images?

**Comment by ivy94419** [comment]
@ruotianluo No，for simplify,  I selected a part of 1000 val images and test them on your pretrained model and my model. The gap of two performance is huge.

**Comment by ruotianluo** [comment]
I'm not sure. I haven't touched fc model for a while. Have you tried attention-based model?

**Comment by ivy94419** [comment]
@ruotianluo You mean show_attend_tell model? I haven't try it. I will try it now. 
All parameters are set as default, but some parameters I still need to confirm, batchsize=10, ss_prob=0.0, att_size=7  (Since att_size=14 results a 368GB feats_att.h5, too big)

**Comment by ruotianluo** [comment]
Try att2in2. att_size 7 is fine.

**Comment by ivy94419** [comment]
@ruotianluo Training att2in2 model also follow the the training method? I mean I only need to run train.py and do not need any steps during training process?

**Comment by ruotianluo** [comment]
Yes. Just change the caption_model arg

**Comment by ivy94419** [comment]
@ruotianluo 
I have trained att2in2 for 14 epochs for now. But the performance is not inspiring.
**Epoch 11 model test result**
![image](https://user-images.githubusercontent.com/4054390/43124041-85c53512-8f58-11e8-8957-9168623e1371.png)
**Epoch 13 model test result**
![image](https://user-images.githubusercontent.com/4054390/43124052-90b37ace-8f58-11e8-866a-d19ef16487fd.png)
**Epoch 13 validation loss** 
![image](https://user-images.githubusercontent.com/4054390/43124096-b0c07cb8-8f58-11e8-80fd-7800a0fc2e84.png)
**Epoch 14 model test result**
![image](https://user-images.githubusercontent.com/4054390/43124119-beb9c130-8f58-11e8-9050-eda59692d8d5.png)

It seems hard to more improve space.
But I achieved the following scores on att2in2 pretrained model on the same 1000 images
![image](https://user-images.githubusercontent.com/4054390/43124186-f38f38d6-8f58-11e8-8cb0-8cbe9ca7ad2f.png)

**Comment by ruotianluo** [comment]
To be honest, I have no idea.
